backend/api/views/failed_asset_logs.py

```diff
@@ -0,0 +1,41 @@
+from rest_framework import response as drf_response
+from rest_framework import status
+from rest_framework import viewsets
+from rest_framework.decorators import action
+
+from api.errors import AssetPermissionError
+from api.models import ComputeTask
+from api.models import Function
+from api.views import utils as view_utils
+from substrapp.models import asset_failure_report
+
+
+class FailedAssetLogsViewSet(view_utils.PermissionMixin, viewsets.GenericViewSet):
+    queryset = asset_failure_report.AssetFailureReport.objects.all()
+
+    @action(detail=True, url_path=asset_failure_report.LOGS_FILE_PATH)
+    def file(self, request, pk=None) -> drf_response.Response:
+        report = self.get_object()
+        channel_name = view_utils.get_channel_name(request)
+        if report.asset_type == asset_failure_report.FailedAssetKind.FAILED_ASSET_FUNCTION:
+            asset_class = Function
+        else:
+            asset_class = ComputeTask
```

Comment:
Nitpick (especially given we don't have much time), but couldn't/shouldn't we have a FailedAssetKind -> Model mapping to avoid this else that considers anything not a function is a CP?

Reply:
Maybe worth opening not to forget about this suggestion in the refactoring?

Reply:
Agree with Sarah, as internal errors are still considered to be part of Compute tasks